wundergraph_cli/src/print_schema/mod.rs

```diff
@@ -311,41 +311,51 @@ mod tests {
         std::thread::sleep(std::time::Duration::from_secs(1));
 
         let query = "{\"query\": \"{ Users { id  name  } } \"}";
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(query)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
-
         let mutation = r#"{"query":"mutation CreateUser {\n  CreateUser(NewUser: {name: \"Max\"}) {\n    id\n    name\n  }\n}","variables":null,"operationName":"CreateUser"}"#;
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(mutation)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
+        let t1 = request_test(&client, &listen_url, query);
+        let t2 = request_test(&client, &listen_url, mutation);
+        let t3 = request_test(&client, &listen_url, query);
 
-        let mut r = client
+        child.kill().unwrap();
+        child.wait().unwrap();
+
+        if !t1 || !t2 || !t3 {
+            panic!("round_trip failed")
+        }
+    }
+
+    fn request_test(client: &reqwest::Client, listen_url: &str, body: &'static str) -> bool {
+        let r = client
             .post(&format!("http://{}/graphql", listen_url))
-            .body(query)
+            .body(body)
             .header(
                 reqwest::header::CONTENT_TYPE,
                 reqwest::header::HeaderValue::from_static("application/json"),
             )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
+            .send();
 
-        child.kill().unwrap();
-        child.wait().unwrap();
+        let nok = |e| {
+            if cfg!(feature = "debug") {
+                println!("{}", e)
+            }
+            false
+        };
+
+        match r {
+            Ok(mut r) => {
+                let builder = std::thread::Builder::new().name("round_trip".into());
```

Comment:
It's better to use `std::panic::catch_unwind()` instead here in my opinion. Also this would allow us to just bubble up the "error" in the fail case.

Reply:
4152b3d55aa77d43fe5acaf134952eee21143dfa contains `std::panic::catch_unwind` based approach